Processes

VeraSimon opened the pull request:
- [x] ex1 (fork)
- [x] ex2 (fopen)
- [x] ex3 (wait)
- [x] ex4 (exec)
- [x] ex5 (pipe)
- [x] ex6 (clock_gettime)

VeraSimon committed:
ex1
Played around with fork()

VeraSimon committed:
ex2
Opened and wrote to a file from two different processes!

VeraSimon committed:
ex3
Used wait() to ensure console print order of a parent and child process

LuisanSuarez commented on ex3/ex3.c:
```diff
@@ -10,6 +10,23 @@
+    else
+    {
+        // Parent
+        int wc = waitpid(fark, NULL, 0);
```
perfect

LuisanSuarez commented on ex2/text.txt:
```diff
@@ -0,0 +1,2 @@
+Hello chilld. This is your forthar 2891.
```
nice 👍

LuisanSuarez commented on ex2/ex2.c:
```diff
@@ -1,14 +1,37 @@
 #include <unistd.h>
 #include <stdlib.h>
 
 int main(void)
```
looks good!

LuisanSuarez commented on ex1/ex1.c:
```diff
@@ -2,13 +2,50 @@
+        printf("Parent after setting x: %i\n", x);
+    }
+
+    printf("Post-fork x: %i, from PID: %i\n", x, (int)getpid());
```
nice going with this statement here. it makes super clear that these are separate processes entirely

VeraSimon committed:
ex4
Played around with the exec family after doing some extra reading on the topic.

VeraSimon committed:
ex5
Created a pipe between parent and child. Messages were passed. Worlds were greeted.

VeraSimon committed:
ex6
Played with the clock. Times some stuff.

LuisanSuarez reviewed (approved):
.

LuisanSuarez commented on ex6/ex6.c:
```diff
@@ -21,6 +21,34 @@ and `clock_gettime()` should work just fine.
+
+        // calculate the time difference and add the runtime to runtimes
+        //     |--convert seconds to nanoseconds---| + trailing nanoseconds
+        diff = BILLION * (end.tv_sec - start.tv_sec) + end.tv_nsec - start.tv_nsec;
```
👍

LuisanSuarez commented on ex5/ex5.c:
```diff
@@ -1,22 +1,58 @@
+    char inbuf[MSGSIZE];
+    int p[2];
+
+    if (pipe(p) < 0)
```
well done

LuisanSuarez commented on ex5/ex5.c:
```diff
@@ -1,22 +1,58 @@
+        int wc = waitpid(fark, NULL, 0);
+        for (int i = 0; i < 3; i++)
+        {
+            read(p[0], inbuf, MSGSIZE);
```
great job :)

LuisanSuarez commented on ex4/ex4.c:
```diff
@@ -1,16 +1,41 @@
+        // Child
+        char *farkingArgs[] = {"ls", "-l", "-a", "-h", NULL};
+        printf("The args: %s\n", farkingArgs);
+        execvp(farkingArgs[0], farkingArgs);
```
nice :)

LuisanSuarez commented:
awesome commit messages, thanks for keeping the code so clean!
